### PR TITLE
Email templates

### PR DESCRIPTION
**Pull request by jasiek-net**

### CODE REVIEW

File: src/helpers/passport.ts [hunk 1]
```diff
@@ -33,7 +32,7 @@ export function authenticate(req: Request, res: Response, next: NextFunction): P
   return new Promise((resolve, reject) => {
     passport.authenticate(
       'jwt',
-      { session: conf.jwt.session },
+      { session: process.env.JWT_SESSION },
```
Comment (michalmikolajczyk) [inline src/helpers/passport.ts:35]: 👍

File: src/mailer/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pass: process.env.MAIL_PASS,
+  },
+};
+const transporter = nodemailer.createTransport(config);
+
+function sendEmail(body: any): Promise<any> {
+  return new Promise((res, rej) => {
+    transporter.sendMail(body, (err, info) => {
+      if (err) return rej(`Email sent error ${err}`);
+      return res(`Email sent successfully ${info}`);
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:19]: nice style

File: src/mailer/index.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  auth: {
+    user: process.env.MAIL_USER,
+    pass: process.env.MAIL_PASS,
+  },
+};
+const transporter = nodemailer.createTransport(config);
+
+function sendEmail(body: any): Promise<any> {
+  return new Promise((res, rej) => {
+    transporter.sendMail(body, (err, info) => {
```
Comment (michalmikolajczyk) [inline src/mailer/index.ts:17]: don't you need to return `transporter.sendMail()` here?

File: src/mailer/templates.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
+export function signinMessage(name: string, token: string, code: string): object {
+  switch (code) {
+    case 'pl':
+      return {
+        subject: `Cześć ${name}, jedź z VIKING GARAGE`,
+        text:
+`Cześć ${name}!
+
+Witaj w VIKING GARAGE!
```
Comment (michalmikolajczyk) [inline src/mailer/templates.ts:9]: to nie z tłumaczeń?

**Review by michalmikolajczyk — CHANGES_REQUESTED** [review]
do sprawdzenia

**Comment by michalmikolajczyk** [comment]
please rebase on `master`

File: src/mailer/index.ts (hunk 3 above)
Comment (jasiek-net) [reply src/mailer/index.ts:17]: nope, I passed callback to `sendMail`, it's not return any `Promise`, I resolved/rejected the `Promise` inside callback.

File: src/mailer/templates.ts (hunk 4 above)
Comment (jasiek-net) [reply src/mailer/templates.ts:9]: firstly I added module with translation, but for now we have only two emails so it's more convenient to translate them like this

**Review by michalmikolajczyk — APPROVED** [review]

**Comment by michalmikolajczyk** [comment]
hey @jonzee can you try to rebase this on `master` please?

**Comment by michalmikolajczyk** [comment]
halo

**Pull request merged**